src/common/tse.c

```diff
@@ -249,6 +249,8 @@ tse_task_decref(tse_task_t *task)
 	D_ASSERT(d_list_empty(&dtp->dtp_dep_list));
 	D_ASSERT(d_list_empty(&dtp->dtp_comp_cb_list));
 
+	dtp->dtp_magic = TASK_BAD_MAGIC;
+
 	/*
 	 * MSC - since we require user to allocate task, maybe we should have
```

Comment:
This comment looks outdated, all task are allocated in this tse_task_create() function in this file.